### PR TITLE
TST: split up test_concat.py #37243 - more follows up

### LINKED ISSUE #37243: TST: split up test_concat.py
we should split this to a sub-module (concat) and then split logically (similar to merge)

https://github.com/pandas-dev/pandas/blob/master/pandas/tests/reshape/test_concat.py

### PR DESCRIPTION
**Pull request by kamilt5**
* created test_categorical.py

- [x] closes #37243 
- [ ] tests added / passed
- [x] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by laffite56** [comment]
I will finish it tomorrow, i didn't have much free time today. There is not much left, i have to create test_sort and test_invalid and make some small changes.

**Comment by laffite56** [comment]
most of the work should be done, all tests are working, it was 291 passed and 1 xfailed before and it's the same now.

**Comment by laffite56** [comment]
I haven't touched test_append_common and test_append, but these files looks ok for me.

File: pandas/tests/reshape/concat/test_categorical.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+import pytest
+
+from pandas.core.dtypes.dtypes import CategoricalDtype
+
+import pandas as pd
+from pandas import Categorical, DataFrame, Series
+import pandas._testing as tm
+
+
+@pytest.fixture(params=[True, False])
```
Comment (jreback) [inline pandas/tests/reshape/concat/test_categorical.py:11]: i would move fixtures to a concat/conftest.py (unless they are truly specific to a particular file)

File: pandas/tests/reshape/concat/test_empty.py [hunk 2]
```diff
@@ -0,0 +1,257 @@
+import numpy as np
+import pytest
+
+import pandas as pd
+from pandas import DataFrame, Index, Series, concat, date_range
+import pandas._testing as tm
+
+
+@pytest.fixture(params=[True, False])
```
Comment (jreback) [inline pandas/tests/reshape/concat/test_empty.py:9]: yeah so this is for sure a common fixture

**Review by jreback — CHANGES_REQUESTED** [review]
can you show the before / after tests counts (they should match).

make the change in conftest.py and i think this should be able to be green if you merge master.

**Comment by laffite56** [comment]
Before: 
obraz

After: 
obraz

File: pandas/tests/reshape/concat/test_series.py [hunk 3]
```diff
@@ -52,7 +46,7 @@ def test_concat_empty_and_non_empty_series_regression(self):
         result = pd.concat([s1, s2])
         tm.assert_series_equal(result, expected)
 
-    def test_concat_series_axis1(self, sort=sort):
+    def test_concat_series_axis1(self, sort=True):
         ts = tm.makeTimeSeries()
```
Comment (laffite56) [inline pandas/tests/reshape/concat/test_series.py]: I don't know about this, after moving sort to conftest.py, this test was giving me error, so i have changed default to True. Is this good, or do i have to move fixture to this file and revert this change?

Comment (jreback) [reply pandas/tests/reshape/concat/test_series.py]: pls revert this, you should not be changing *any* code (literally just removing the fixture definition itself)

File: pandas/tests/reshape/concat/test_datetimes.py [hunk 4]
```diff
@@ -518,3 +512,15 @@ def test_concat_period_other_series(self):
         result = concat([x, y], ignore_index=True)
         tm.assert_series_equal(result, expected)
         assert result.dtype == "object"
+
+
+def test_concat_timedelta64_block():
+    from pandas import to_timedelta
```
Comment (jreback) [inline pandas/tests/reshape/concat/test_datetimes.py:518]: import at the top

File: pandas/tests/reshape/concat/test_datetimes.py [hunk 5]
```diff
@@ -518,3 +512,15 @@ def test_concat_period_other_series(self):
...
+
+    rng = to_timedelta(np.arange(10), unit="s")
+
+    df = DataFrame({"time": rng})
+
+    result = concat([df, df])
+    assert (result.iloc[:10]["time"] == rng).all()
```
Comment (jreback) [inline pandas/tests/reshape/concat/test_datetimes.py:525]: ca n you this use tm.assert_frame_equal instead (and use a fully constructed dataframe)

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
couple of comments, ping on green.

File: pandas/tests/reshape/concat/test_series.py (hunk 3 above)
Comment (laffite56) [reply pandas/tests/reshape/concat/test_series.py]: ok, changed.

**Comment by laffite56** [comment]
ok, i will check it in a moment.

**Comment by laffite56** [comment]
@jreback everything done

**Comment by laffite56** [comment]
Can I ask what does "ping on green" mean? Is it the green "review changes" button in "files changed" section? Or is it just typing comment, so you can see new activity?

**Review by kamilt5 — COMMENTED** [review]
green

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @laffite56
